Autoyast migration for powerVM

tinawang123 opened the pull request:
Add new autoyast profile for powerVM upgrade testing, and powerVM
need reboot after install system

- Related ticket: https://progress.opensuse.org/issues/71935
- Verification run:
https://openqa.suse.de/tests/5341502#
Not impact others' case: https://openqa.suse.de/tests/5341504#

lemon-suse commented on data/autoyast_sle15/autoyast_scc_up_powervm.xml:
```diff
@@ -0,0 +1,1159 @@
+    </user>
+    <user t="map">
+      <encrypted t="boolean">true</encrypted>
+      <fullname>Postfix Daemon</fullname>
```
Need we configure the postfix daemon?

tinawang123 replied:
Just follow function's autoyast profile.

rwx788 commented on tests/autoyast/installation.pm:
```diff
@@ -138,7 +138,7 @@ sub run {
     # Autoyast reboot automatically without confirmation, usually assert 'bios-boot' that is not existing on zVM
     # So push a needle to check upcoming reboot on zVM that is a way to indicate the stage done
-    push @needles, 'autoyast-stage1-reboot-upcoming' if check_var('ARCH', 's390x') || is_pvm;
+    push @needles, 'autoyast-stage1-reboot-upcoming' if check_var('ARCH', 's390x') || (is_pvm && !is_upgrade);
```
Hmm, I'm not sure why we need this change, as this reboot is required to handle second stage. From you VR it looks like there is no second stage, but in any case this should not be backend specific. For such cases, I would suggest to handle it later and exit before second stage loop, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11844/files#diff-269577be56cf62326b2be1ff7b71693c5b5aafd0cf37a5ebf7492d63bb9a0245R332

rwx788 replied:
After looking at your code, seems that we can improve it for the autoyast installations on powerVM, as in our case we don't have this pop-up (https://openqa.suse.de/tests/5341502#step/installation/9), so we cannot rely on it: https://openqa.suse.de/tests/5337658
Do you know what causes this behavior? In any case, as stated above, I believe that it is same behavior on s390x too, so is not powerVM specific.

tinawang123 replied:
If I don't change this part. It cannot restart system: https://openqa.suse.de/tests/5341423#step/installation/10. It seems it didn't finish to install system then it reboot system.  I don't know why it will pop reboot screen. So I just change it for upgrade cases.

rwx788 replied:
I guess it's due to `reboot_timeout=0` boot option, which we can use in all scenarios. As of now this change is safe, but in general handling here likely depends on that boot option and not just backend. Same can be applied for s390x.

lemon-suse replied:
I think it is really different here for s390x and power, since it hasn't finished installation for power so skip here is reasonable. @rwx788 WDYT?

rwx788 replied:
As per comment from @tinawang123 in this PR, seems that this behavior depends on the settings in the profile, which is more realistic. I will check myself if we can apply this solution for complex cases like s390x and powerVM, as those platforms have a challenge to detect when reboot will happen in order to successfully reconnect to them.
As stated before, this change is safe and won't break other tests, even though we have confirmed that it's not about powerVM here, but rather about the profile. So now you will need to assure that profile contains `security` section defined as proposed here.

rwx788 reviewed (commented):
Feels like the profile you have added is based on the generated one, as it's very big. I would suggest to remove unnecessary parts from it, which will simplify maintenance of the profile in future and keep only parts which are relevant for the upgrade scenario.

tinawang123 commented:
> Feels like the profile you have added is based on the generated one, as it's very big. I would suggest to remove unnecessary parts from it, which will simplify maintenance of the profile in future and keep only parts which are relevant for the upgrade scenario.

Actually, I don't very familiar with all elements of autoyast profile.  I just copy this file from function cases and then change 'suse_register', 'software' and 'upgrade' those three parts. Others I don't change just follow the original profile file.

rfan1 commented:
> > Feels like the profile you have added is based on the generated one, as it's very big. I would suggest to remove unnecessary parts from it, which will simplify maintenance of the profile in future and keep only parts which are relevant for the upgrade scenario.
> 
> Actually, I don't very familiar with all elements of autoyast profile. I just copy this file frHom function cases and then change 'suse_register', 'software' and 'upgrade' those three parts. Others I don't change just follow the original profile file.

Hi @tinawang123, in a simple way, you can install a vm/powervm machine by yourself based on your test requirement, and then generate an new autoyast file. but in general, we can do ~everything we want to in a xml file instead of adding a new needle.

tinawang123 commented:
Updated autoyast profile
Verified: https://openqa.suse.de/tests/5350568
But it still has reboot screen.

tinawang123 commented:
Let me explain the situation:
1. Use function team's autoyast profile, need add 'upgrade' element at profile. Then my case will pop 'reboot' screen (Refer: https://openqa.suse.de/tests/5341502#), but function's case which don't use 'upgrade' element, does not have 'reboot'. (Refer:  https://openqa.suse.de/tests/5341504#)
2. If use powerKVM autoyast profile, the system cannot reboot successfully (Refer:  https://openqa.suse.de/tests/5341423). I need add 'security' element which have console reboot. Got the result: add 'security' element for powerVM case, it has reboot screen (Refer: https://openqa.suse.de/tests/5350568). If don't have 'security' element , powerVM case cannot restart successfully.
My summarized: Use 'Upgrade' and 'security' element at powerVM autoyast profile, it will generate reboot screen. But those two elements must be used at powerVM autoyast migration profile, or the case will fail. So at 'installation' test code, I add needle to send 'ret' to reboot screen.

rwx788 commented:
> Let me explain the situation:
> 
>     1. Use function team's autoyast profile, need add 'upgrade' element at profile. Then my case will pop 'reboot' screen (Refer: https://openqa.suse.de/tests/5341502#), but function's case which don't use 'upgrade' element, does not have 'reboot'. (Refer:  https://openqa.suse.de/tests/5341504#)
> 
>     2. If use powerKVM autoyast profile, the system cannot reboot successfully (Refer:  https://openqa.suse.de/tests/5341423). I need add 'security' element which have console reboot. Got the result: add 'security' element for powerVM case, it has reboot screen (Refer: https://openqa.suse.de/tests/5350568). If don't have 'security' element , powerVM case cannot restart successfully.
>        My summarized: Use 'Upgrade' and 'security' element at powerVM autoyast profile, it will generate reboot screen. But those two elements must be used at powerVM autoyast migration profile, or the case will fail. So at 'installation' test code, I add needle to send 'ret' to reboot screen.

Thanks for sharing this information, this still sounds a bit obscure to me as is not mentioned in the official documentation. As it simplifies handling of powerVM, and potentially s390x installations, we can agree to add this settings to all AY installations instead of currently existing workaround.
In any case it should not block this PR from merging as we can act on this in parallel.
Thanks again for the details you have shared here!

rwx788 reviewed (commented):
FYI, I've added same config as you have:
```
    <security>
        <console_shutdown>reboot</console_shutdown>
        <disable_restart_on_update>no</disable_restart_on_update>
        <disable_stop_on_removal>no</disable_stop_on_removal>
        <extra_services>insecure</extra_services>
        <hibernate_system>active_console</hibernate_system>
    </security>
```
To one of our fresh installations and did NOT get reboot pop-up. So indeed it seems to be upgrade specific only. I will talk to YaST team tomorrow to be more sure what is expected.

As for your changes, we can keep it as it is for now, or you can remove/change `security` section not to modify `tests/autoyast/installation.pm`.

tinawang123 commented:
> FYI, I've added same config as you have:
> 
> ```
>     <security>
>         <console_shutdown>reboot</console_shutdown>
>         <disable_restart_on_update>no</disable_restart_on_update>
>         <disable_stop_on_removal>no</disable_stop_on_removal>
>         <extra_services>insecure</extra_services>
>         <hibernate_system>active_console</hibernate_system>
>     </security>
> ```
> 
> To one of our fresh installations and did NOT get reboot pop-up. So indeed it seems to be upgrade specific only. I will talk to YaST team tomorrow to be more sure what is expected.
> 
> As for your changes, we can keep it as it is for now, or you can remove/change `security` section not to modify `tests/autoyast/installation.pm`.

This part is which I copied from function cases' autoyast profile. For function cases it don't have reboot screen with those elements. But for migration case, as I also added 'upgrade' element, it will show reboot screen. And for powerVM autoyast migration, I have to use 'security' section to reboot console. As I use 'security' and 'upgrade' section at my autoyast profile, it will show reboot screen.

rwx788 commented:
I've talked to YaST team, and they have confirmed that this pop-up is not expected. And if, by default, there would be no timeout on this pop-up it would be a sever bug, as AY installations should work without interactions from the user side. I will take a look in the code to see what causes such behavior in case of upgrade and if we can have control over it.

tinawang123 commented:
> I've talked to YaST team, and they have confirmed that this pop-up is not expected. And if, by default, there would be no timeout on this pop-up it would be a sever bug, as AY installations should work without interactions from the user side. I will take a look in the code to see what causes such behavior in case of upgrade and if we can have control over it.

I opened one bug for this problem: https://bugzilla.suse.com/show_bug.cgi?id=1181625

lemon-suse commented:
Since you have filed a bug for this unexpected popup message, please add soft-failure record when this tag matched for 'autoyast-boot' to alert others here is a bug for power.

tinawang123 commented:
> Since you have filed a bug for this unexpected popup message, please add soft-failure record when this tag matched for 'autoyast-boot' to alert others here is a bug for power.

As it only impact one case, if I add softfail, it will make many other cases become yellow. If this bug fixed, I will remove the condition code.  Thanks for your suggestion.

lemon-suse commented:
LGTM， please keep following the status of the bug, and update code if needed.
